### PR TITLE
[ace] 6.5.10

### PR DESCRIPTION
**Pull request by jwillemsen**
Upgrade ACE to 6.5.10

### CODE REVIEW

**Comment by jwillemsen** [comment]
Patch was removed because it is now part of ACE 6.5.10.

File: ports/ace/portfile.cmake [hunk 1]
```diff
@@ -5,41 +5,39 @@ if("tao" IN_LIST FEATURES)
   if(VCPKG_TARGET_IS_WINDOWS)
       # Don't change to vcpkg_from_github! This points to a release and not an archive
       vcpkg_download_distfile(ARCHIVE
-          URLS "https://github.com/DOCGroup/ACE_TAO/releases/download/ACE%2BTAO-6_5_9/ACE%2BTAO-src-6.5.9.zip"
-          FILENAME ACE-TAO-6.5.9.zip
-          SHA512 de626d693911ea6b43001b16183996bd537777b42530a95ef226265948802b87aaac935d92265f1dda39c864a875d669a10cdcb0083e3dc7c3f5f661a5ee9d79
+          URLS "https://github.com/DOCGroup/ACE_TAO/releases/download/ACE%2BTAO-6_5_10/ACE%2BTAO-src-6.5.10.zip"
```
Comment (JackBoosY) [inline ports/ace/portfile.cmake:8]: We should use 2 macros.

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/ace/portfile.cmake (hunk 1 above)
Comment (jwillemsen) [reply ports/ace/portfile.cmake:8]: Do you have a suggestion which macro? I only wanted to update the files to a new release

Comment (jwillemsen) [reply ports/ace/portfile.cmake]: Add macros

File: ports/ace/portfile.cmake [hunk 2]
```diff
@@ -1,45 +1,46 @@
 # Using zip archive under Linux would cause sh/perl to report "No such file or directory" or "bad interpreter"
 # when invoking `prj_install.pl`.
 # So far this issue haven't yet be triggered under WSL 1 distributions. Not sure the root cause of it.
+set(VERSION 6.5.10)
```
Comment (JackBoosY) [inline ports/ace/portfile.cmake:4]:
```suggestion
set(ACE_VERSION 6.5.10)
```

File: ports/ace/portfile.cmake [hunk 3]
```diff
@@ -1,45 +1,46 @@
...
+set(VERSION_DIRECTORY 6_5_10)
```
Comment (JackBoosY) [inline ports/ace/portfile.cmake:5]:
```suggestion
set(ACE_VERSION_DIRECTORY 6_5_10)
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Review by JackBoosY — APPROVED** [review]

**Comment by JackBoosY** [comment]
Let my colleague check this PR.

File: ports/ace/portfile.cmake [hunk 4]
```diff
@@ -1,45 +1,46 @@
 # Using zip archive under Linux would cause sh/perl to report "No such file or directory" or "bad interpreter"
 # when invoking `prj_install.pl`.
 # So far this issue haven't yet be triggered under WSL 1 distributions. Not sure the root cause of it.
+set(ACE_VERSION 6.5.10)
+set(ACE_VERSION_DIRECTORY 6_5_10)
```
Comment (c72578) [inline ports/ace/portfile.cmake:5]: How about:
```cmake
string(REPLACE "." "_" ACE_VERSION_DIRECTORY ${ACE_VERSION})
```

Comment (jwillemsen) [reply ports/ace/portfile.cmake]: thanks for the suggestion, made that change

**Comment by ras0219-msft** [comment]
LGTM, thanks for the update!

**Comment by JackBoosY** [comment]
All test passed on my machine. I just finished test.

**Pull request merged**